### PR TITLE
added info messages and requests to enable cam and mic

### PR DESCRIPTION
**Pull request by ManukMamyan**

### CODE REVIEW

File: src/SipConnector.ts [hunk 1]
```diff
@@ -494,6 +507,22 @@ export default class SipConnector {
     delete this.incomingSession;
   };
 
+  enableMainCam(): Promise<void> {
```
Comment (Krivega) [inline src/SipConnector.ts:510]: askPermissionToEnableCam

File: src/SipConnector.ts [hunk 2]
```diff
@@ -494,6 +507,22 @@ export default class SipConnector {
...
+    const extraHeaders = [HEADER_ENABLE_MAIN_CAM];
+
+    return this.session!.sendInfo(CONTENT_TYPE_MAIN_CAM, undefined, {
+      extraHeaders,
+    });
+  }
+
+  enableMic(): Promise<void> {
```
Comment (Krivega) [inline src/SipConnector.ts:518]: askPermissionToEnableMic

File: src/SipConnector.ts [hunk 3]
```diff
@@ -144,6 +152,11 @@ type TChannels = {
   outputChannels: string;
 };
 
+type TMediaState = {
+  mainCam: '0' | '1';
```
Comment (Krivega) [inline src/SipConnector.ts:156]: boolean

File: src/__tests__/mediaState.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    await sipConnector.connect(dataForConnectionWithAuthorization);
+    await sipConnector.call({ number, mediaStream });
+
+    mockFn = jest.fn(() => {
+      return undefined;
+    });
+
+    // @ts-ignore
+    sipConnector.session.sendInfo = mockFn;
```
Comment (Krivega) [inline src/__tests__/mediaState.ts:34]: add function to Session mock

**Review by Krivega — CHANGES_REQUESTED** [review]

File: src/SipConnector.ts (hunk 3 above)
Comment (ManukMamyan) [reply src/SipConnector.ts:156]: +

File: src/SipConnector.ts (hunk 2 above)
Comment (ManukMamyan) [reply src/SipConnector.ts:518]: +

File: src/SipConnector.ts (hunk 1 above)
Comment (ManukMamyan) [reply src/SipConnector.ts:510]: +

File: src/__tests__/mediaState.ts (hunk 4 above)
Comment (ManukMamyan) [reply src/__tests__/mediaState.ts:34]: will change in later iteration

File: src/SipConnector.ts [hunk 5]
```diff
@@ -1428,6 +1457,19 @@ export default class SipConnector {
     this.session!.sendInfo(CONTENT_TYPE_CHANNELS, undefined, { extraHeaders });
   }
 
+  sendMediaState({ mainCam, mic }: TMediaState) {
```
[the remark this answers is not in the document]
Comment (ManukMamyan) [reply src/SipConnector.ts]: +

**Pull request merged**